src/dispatch/DataPlaneClient.ts

```diff
@@ -65,20 +64,20 @@ export class DataPlaneClient {
     public sendFetch = async (
         putRumEventsRequest: PutRumEventsRequest
     ): Promise<{ response: HttpResponse }> => {
-        const host = getHost(this.config.endpoint);
         const serializedRequest: string = JSON.stringify(
             serializeRequest(putRumEventsRequest)
         );
+        const path = this.config.endpoint.pathname.replace(/\/$/, '');
```

Comment:
consider getting these values from config on line 67 like so:

```ts
const { pathname, host, hostname, protocol } = this.config.endpoint
```

Makes the code following a bit cleaner

Reply:
Agreed

Reply:
I'm not a big fan of destructuring. In many cases its additional layer of abstraction makes the code less clear (especially for those new to JS/TS). E.g., is the value from the config, or from a local variable? I guess it saves some bits, but there is lower hanging fruit if we're worried about size.

Reply:
In my mind it is easier to parse -- I agree w you on the array destructuring, but for objects, I still maintain that it makes sense.